src/app/shared/components/hero/hero.component.spec.ts

```diff
@@ -1,37 +1,45 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { HeroComponent } from './hero.component';
-import { HeroComponentVM } from './hero.component.vm';
+import { HeroVM } from './hero-vm.model';
 import { Component, signal } from '@angular/core';
+import { getTranslocoModule } from 'transloco-testing.module';
+import { TranslocoService } from '@jsverse/transloco';
 
 @Component({
   template: `
-    <app-hero [vm]="vm()">
-      <h1 class="testH1" id="test">The future of irrigation is now</h1>
-      <p class="testP">IRRIOT is a completely wireless precision irrigation automation system</p>
-      <a class="testAnchor" href="">Contact Us</a>
-      <button class="testButton">Test button</button>
-      <div class="testDiv">Should not be projected</div>
-    </app-hero>
+    @if (vm(); as vm) {
+      <app-hero [vm]="vm">
+        <h1 class="testH1" id="test">{{ vm.titles[0] | transloco }}</h1>
+        <p class="testP">{{ vm.titles[1] }}</p>
+        <a class="testAnchor" href="">{{ vm.buttonTitles[0] | transloco }}</a>
+        <button class="testButton">{{ vm.buttonTitles[1] | transloco }}</button>
+        <div class="testDiv">Should not be projected</div>
+      </app-hero>
+    }
   `,
 })
 class TestHostComponent {
-  vm = signal<HeroComponentVM | null>({
+  vm = signal<HeroVM>({
     backgroundImageSrc: 'assets/farming.jpg',
+    titles: ['LandingPage.hero.0.title', 'LandingPage.hero.1.title'],
+    buttonTitles: ['LandingPage.hero.0.buttonTitle', 'LandingPage.hero.1.buttonTitle'],
   });
 }
 describe('HeroComponent', () => {
   let component: TestHostComponent;
   let fixture: ComponentFixture<TestHostComponent>;
   let compiled: HTMLElement;
+  let translocoService: TranslocoService;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [HeroComponent],
+      imports: [HeroComponent, getTranslocoModule()],
```

Comment:
You could easily pass in the the VM and a random translation key to check whenever the translation is working and whenever it is in the correct place.

```ts
import { provideTranslocoTestingModule } from '@ngneat/transloco';

const en =  { title1: 'Hello', title2: 'World', buttonTitle1: '123', buttonTitle2: '234' }

beforeEach(() => {
  TestBed.configureTestingModule({
    imports: [
      provideTranslocoTestingModule({
        langs: { en },
        translocoConfig: { availableLangs: ['en'], defaultLang: 'en' },
      })
    ],
  });
});
...
class TestHostComponent {
  vm = signal<HeroVM>({
    backgroundImageSrc: 'assets/farming.jpg',
    titles: ['title1', 'title2'],
    buttonTitles: ['buttonTitle1', 'buttonTitle2'],
  });
```

This makes it testable as every resource contained within the test and there is no external dependency (`en.json` or `hu.json`). So you can write a test where you check if the text within the element is the same as in the object you passed in.

```ts
expect(elementTitle1.innerHTML).toBe(en.title1)
```